pkg/common/utils.go

```diff
@@ -479,12 +474,49 @@ func existingErrorCode(err error) (codes.Code, error) {
 	if err == nil {
```

Comment:
Can you add another comment explaining what existingErrorCode does as a whole? I still think its unclear why we 
need to do !errors.As(err, &googleErr)? I think this would be a good comment: 

existingErrorCode returns the existing gRPC Status error code for the given error, if one exists, or Unknown if one doesn't exist. Since github.com/googleapis/gax-go/v2/apierror now wraps googleapi errors (returned from GCE API calls), and sets their status error code to Unknown, we now have to make sure we only return existing error codes from errors that are either TemporaryErrors, or errors that do not wrap googleAPI errors. Otherwise, we will return Unknown for all GCE API calls that return googleapi errors.